lib/actions/user.js

```diff
@@ -134,56 +158,116 @@ export function createOrUpdateUser (userData) {
+    const { id } = tripData
+    let requestUrl, method
 
-    if (otpMiddleware) {
-      const { accessToken } = user
+    // Determine URL and method to use.
+    if (isNew) {
+      requestUrl = `${apiBaseUrl}${API_MONITORTRIP_PATH}`
+      method = 'POST'
+    } else {
+      requestUrl = `${apiBaseUrl}${API_MONITORTRIP_PATH}/${id}`
+      method = 'PUT'
+    }
 
-      let result
-      if (isNew) {
-        result = await addTrip(otpMiddleware, accessToken, tripData)
-      } else {
-        result = await updateTrip(otpMiddleware, accessToken, tripData)
+    const { data, message, status } = await secureFetch(requestUrl, accessToken, apiKey, method, {
+      body: JSON.stringify(tripData)
+    })
+
+    // TODO: improve the UI feedback messages for this.
+    if (status === 'success' && data) {
+      if (!silentOnSuccess) {
+        alert('Your preferences have been saved.')
       }
 
-      // TODO: improve the UI feedback messages for this.
-      if (result.status === 'success' && result.data) {
-        if (!silentOnSuccess) {
-          alert('Your preferences have been saved.')
-        }
+      // Reload user's monitored trips after add/update.
+      await dispatch(fetchUserMonitoredTrips(accessToken))
+
+      // Finally, navigate to the saved trips page.
+      dispatch(routeTo('/savedtrips'))
+    } else {
+      alert(`An error was encountered:\n${JSON.stringify(message)}`)
+    }
+  }
+}
 
-        // Reload user's monitored trips after add/update.
-        await dispatch(fetchUserMonitoredTrips(accessToken))
+/**
+ * Deletes a logged-in user's monitored trip,
+ * then, if that was successful, refreshes the redux monitoredTrips state.
+ */
+export function deleteUserMonitoredTrip (tripId) {
+  return async function (dispatch, getState) {
+    const { accessToken, apiBaseUrl, apiKey } = getMiddlewareVariables(getState())
+    const requestUrl = `${apiBaseUrl}${API_MONITORTRIP_PATH}/${tripId}`
 
-        // Finally, navigate to the saved trips page.
-        dispatch(routeTo('/savedtrips'))
+    const { message, status } = await secureFetch(requestUrl, accessToken, apiKey, 'DELETE')
+    if (status === 'success') {
+      // Reload user's monitored trips after deletion.
+      await dispatch(fetchUserMonitoredTrips(accessToken))
+    } else {
+      alert(`An error was encountered:\n${JSON.stringify(message)}`)
+    }
+  }
+}
+
+/**
+ * Requests a verification code via SMS for the logged-in user.
+ */
+export function requestPhoneVerificationSms (newPhoneNumber) {
+  return async function (dispatch, getState) {
+    const state = getState()
+    const { number, timestamp } = state.user.lastPhoneSmsRequest
+    const now = new Date()
+
+    // Request a new verification code if we are requesting a different number.
+    // or enough time has ellapsed since the last request (1 minute?).
+    // TODO: Should throttling be handled in the middleware?
+    if (number !== newPhoneNumber || timestamp + 60000 >= now) {
```

Comment:
If it should be, then there should be a way to communicate to a user that they are being throttled. Perhaps it might be easier to handle this in the backend. As it is, there is this hardcoded 60000-millisecond throttle that might need to be updated for each deployment.

Reply:
For now I have added displaying an alert in https://github.com/opentripplanner/otp-react-redux/pull/224/commits/cbf2d86940c5b4c51d9eb03969c22926c1a2ba7b.